lib/ret/avatar.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Ret.Avatar.AvatarSlug do
+  use EctoAutoslugField.Slug, from: :name, to: :slug
+
+  def get_sources(_changeset, _opts) do
+    [:avatar_sid, :name]
+  end
+end
+
+defmodule Ret.Avatar do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Ret.{Avatar, Repo, OwnedFile, Account, Sids}
+  alias Ret.Avatar.{AvatarSlug}
+
+  @schema_prefix "ret0"
+  @primary_key {:avatar_id, :id, autogenerate: true}
+
+  schema "avatars" do
+    field(:avatar_sid, :string)
+    field(:slug, AvatarSlug.Type)
+    belongs_to(:parent_avatar, Avatar, references: :avatar_id)
+
+    field(:name, :string)
+    field(:description, :string)
+    field(:attributions, :map)
+
+    field(:allow_remixing, :boolean)
+    field(:allow_promotion, :boolean)
+    belongs_to(:account, Account, references: :account_id)
+
+    belongs_to(:gltf_owned_file, OwnedFile, references: :owned_file_id)
+    belongs_to(:bin_owned_file, OwnedFile, references: :owned_file_id)
+
+    belongs_to(:base_map_owned_file, OwnedFile, references: :owned_file_id)
+    belongs_to(:emissive_map_owned_file, OwnedFile, references: :owned_file_id)
+    belongs_to(:normal_map_owned_file, OwnedFile, references: :owned_file_id)
+    belongs_to(:orm_map_owned_file, OwnedFile, references: :owned_file_id)
+
+    field(:state, Avatar.State)
+
+    timestamps()
+  end
+
+  def load_parents(avatar, preload_fields \\ [])
+
+  def load_parents(%Avatar{parent_avatar: nil} = avatar, preload_fields),
+    do: avatar |> Repo.preload(preload_fields)
+
+  def load_parents(%Avatar{parent_avatar: %Ecto.Association.NotLoaded{}} = avatar, preload_fields) do
```

Comment:
i think we can drop the `{ parent_avatar: ` match here -- that `NotLoaded` struct might be subject to change, and i think it's still semantically the same fall through to just ignore what non-nil value `parent_avatar` is taking on here